vendors/microchip/boards/ecc608a_plus_winsim/aws_demos/application_code/aws_demo_logging.c

```diff
@@ -577,31 +587,37 @@ static void prvLoggingPrintf( uint8_t usLoggingLevel,
         {
             configASSERT( xLogStreamBuffer );
 
-            /* How much space is in the buffer? */
-            xLength2 = uxStreamBufferGetSpace( xLogStreamBuffer );
-
-            /* There must be enough space to write both the string and the length of
-             * the string. */
-            if( xLength2 >= ( xLength + sizeof( xLength ) ) )
+            /* First write in the length of the data, then write in the data
+             * itself.  Raising the thread priority is used as a critical section
+             * as there are potentially multiple writers.  The stream buffer is
+             * only thread safe when there is a single writer (likewise for
+             * reading from the buffer). */
+            xCurrentTask = GetCurrentThread();
+            iOriginalPriority = GetThreadPriority( xCurrentTask );
+            SetThreadPriority( GetCurrentThread(), THREAD_PRIORITY_TIME_CRITICAL );
             {
-                /* First write in the length of the data, then write in the data
-                 * itself.  Raising the thread priority is used as a critical section
-                 * as there are potentially multiple writers.  The stream buffer is
-                 * only thread safe when there is a single writer (likewise for
-                 * reading from the buffer). */
-                xCurrentTask = GetCurrentThread();
-                iOriginalPriority = GetThreadPriority( xCurrentTask );
-                SetThreadPriority( GetCurrentThread(), THREAD_PRIORITY_TIME_CRITICAL );
-                uxStreamBufferAdd( xLogStreamBuffer,
-                                   0,
-                                   ( const uint8_t * ) &( xLength ),
-                                   sizeof( xLength ) );
-                uxStreamBufferAdd( xLogStreamBuffer,
-                                   0,
-                                   ( const uint8_t * ) cOutputString,
-                                   xLength );
-                SetThreadPriority( GetCurrentThread(), iOriginalPriority );
+                /* How much space is in the buffer? */
+                xLength2 = uxStreamBufferGetSpace( xLogStreamBuffer );
```

Comment:
Might be better to use a different variable for this instead of `xLength2`. Something like `xStreamBufferSpace`.

Reply:
You're right, it looks as if I wanted to economise on the use of local variables! Now changed to `uxSpace`.